pkg/worker/k8srunner.go

```diff
@@ -409,30 +480,59 @@ func (r k8sStepRunner) continueInitContainer(podName string) error {
 	return nil
 }
 
-func (r k8sStepRunner) copyDirToPod(ctx context.Context, destPath, namespace, podName, containerName string) error {
-	// Based on: https://stackoverflow.com/a/57952887
+func (r k8sStepRunner) copyDirToPod(ctx context.Context, destPath string) error {
 	tarReader, err := r.repoTar.Open()
 	if err != nil {
 		return err
 	}
 	defer tarReader.Close()
 
-	reader, writer := io.Pipe()
-	defer reader.Close()
-	defer writer.Close()
 	args := []string{"tar", "-xf", "-", "-C", destPath}
-	exec, err := execInPodPipedStdin(r.RestConfig, namespace, podName, containerName, args)
+	return r.copyToPodStdin(ctx, tarReader, args)
+}
+
+func (r k8sStepRunner) copyDockerfileToPod(ctx context.Context, step wharfyml.StepDocker) error {
+	path := filepath.Join(r.CurrentDir, step.File)
+	if !validateNoParentDirAccess(path) {
+		return fmt.Errorf("%w: %q", errParentDirAccessNotAllowed, path)
+	}
+
+	destPath := filepath.Join(commonRepoVolumeMount.MountPath, step.File)
```

Comment:
Actually think we want to use regular [`path`](https://pkg.go.dev/path) package instead of [`path/filepath`](https://pkg.go.dev/path/filepath) here, as it would otherwise break for Windows users where it would add backslashes.

I deem it OK to hardcode that the Docker image is running on Linux. But the user might not be.

The above declaration of `path := ` should use `filepath` though as it's referncing local files.